lib/auth/oidc_test.go

```diff
@@ -74,10 +75,40 @@ func (s *OIDCSuite) SetUpSuite(c *check.C) {
 		SkipPeriodicOperations: true,
 	}
 	s.a, err = NewServer(authConfig)
-	c.Assert(err, check.IsNil)
+	require.NoError(t, err)
+	return &s
 }
 
-func (s *OIDCSuite) TestCreateOIDCUser(c *check.C) {
+// createInsecureOIDCClient creates an insecure client for testing.
+func createInsecureOIDCClient(t *testing.T, connector types.OIDCConnector) *oidc.Client {
+	conf := oidcConfig(connector)
+	conf.HTTPClient = &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				InsecureSkipVerify: true,
+			},
+		},
+	}
+	client, err := oidc.NewClient(conf)
+	require.NoError(t, err)
+
+	ctx, cancel := context.WithTimeout(context.Background(), defaults.WebHeadersTimeout)
+	defer cancel()
+
+	go func() {
+		defer cancel()
+		client.SyncProviderConfig(connector.GetIssuerURL())
+	}()
+
+	<-ctx.Done()
+	if ctx.Err() != nil {
```

Comment:
I think this check is not necessary since the `Done` channel will be closed so that you can expect a non-nil error:

> [If Done is closed, Err returns a non-nil error explaining why:](https://cs.opensource.google/go/go/+/refs/tags/go1.17.6:src/context/context.go;l=102)